prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -150,43 +153,61 @@ class ActionHistory(
         action.username = userName
     }
 
-    fun filterParameters(request: HttpRequestMessage<String?>): String {
-        // list of values to search and filter
-        val notAllowedHeaderParts = listOf("key", "cookie", "auth")
-        val notAllowedParameterParts = listOf("code")
+    /**
+     * Class captures data about the HTTP request from a sender to the report endpoint
+     *
+     */
+    data class ReceivedReportSenderParameters(
+        val method: HttpMethod,
+        val url: String,
+        @JsonProperty("Headers")
+        val headers: Map<String, String>,
+        @JsonProperty("QueryParameters")
+        val queryParameters: Map<String, String>,
+    ) {
+        companion object {
+            // These header/query params can potentially contain auth information, and we do not want to log them
+            // gitleaks:allow
+            private val parameterNamesToExclude = listOf(
```

Comment:
could save a few cycles by using `setOf` since were iterating through this (albeit short) list for every header.

Reply:
will update